src/Employees/Api/EmployeesController.cs

```diff
@@ -116,4 +117,45 @@ public async Task<ActionResult> UpdateEmergencyContact(string alias, string coun
             return NoContent();
         }
     }
+
+    /**
+     * <returns>A list of the allergies as strings</returns>
+     */
+    [HttpGet("allergies")]
+    [OutputCache(Duration = 60)]
+    public List<string> GetAllergies()
+    {
+        return _employeeService.GetDefaultAllergies().Select(a => a.ToString()).ToList();
+    }
+
+    /**
+     * <returns>A list of the dietary preferences as strings</returns>
+     */
+    [HttpGet("dietaryPreferences")]
+    [OutputCache(Duration = 60)]
+    public List<string> GetDietaryPreferences()
+    {
+        return _employeeService.GetDietaryPreferences().Select(a => a.ToString()).ToList();
+    }
+
+    [Microsoft.AspNetCore.Cors.EnableCors("DashCorsPolicy")]
+    [HttpPost("allergiesAndDietaryPreferences/{country}/{alias}")]
+    public async Task<IActionResult> UpdateAllergiesAndDietaryPreferences(string alias, string country, [FromBody] AllergiesAndDietaryPreferences allergiesAndDietaryPreferences)
+    {
+        var employee = await _employeeService.GetEntityByAliasAndCountry(alias, country);
+
+        if (employee == null)
+        {
+            _logger.LogError("Can't update allergies and dietary preferences because there is no matching Employee to alias {alias} and country {country}", alias, country);
+            return NotFound();
+        }
+        else
+        {
+            await _employeeService.UpdateAllergies(employee, allergiesAndDietaryPreferences.DefaultAllergies, allergiesAndDietaryPreferences.OtherAllergies);
+            await _employeeService.UpdateDietaryPreferences(employee, allergiesAndDietaryPreferences.DietaryPreferences);
+            await _employeeService.AddOrUpdateEmployeeAllergyComment(employee, allergiesAndDietaryPreferences.Comment);

```

Comment:
Her kan du vurdere en `await Task.WhenAll` for å kjøre disse i parallell (om mulig). Ikke nødvendig, men greit å vite om!

Gjelder flere plasser i PRen hvor du gjør await flere linjer på rad - da venter de til den over er ferdig før neste kjøres (sekvensielt).

https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.whenall?view=net-7.0

Reply:
Godt poeng, men gjelder dette også når man henter ut resultatet og bruker det, f.eks. på linje 53 og 54?

Reply:
Dette ble jo borte nå, så ikke relevant lengre 👍 Man kan hente ut resultatet av å awaite tasken etterpå (safest, da er den ferdig og kjøres ikke på nytt) eller bare kjøre på med en .Result (man vet jo at den er ferdig)